modem/design.md

```diff
@@ -0,0 +1,8 @@
+Overview:
+
+![Design diagram](diagram.png)
+
+We create a Transceiver object that will manage the serial connection. It will do command detection and hand off to the appropriate Command object to handle the interaction. The Command object will be subclassed for each type of command we will receive and will handle the processing of the command and generating the response.
```

Comment:
I updated the document to include this. But since all of our commands can be uniquely identified by the first word (text before whitespace) it should be trivial. We could get fancy with a module registration system, (ala REGISTER_MAPPER for MRs), but the command list is small enough it will probably just end up being a if-then tree.

I think the commands should be short lived. I don't see any of them needing to persist state and keeping the objects alive uses extra memory we could use for data handling.

How about SerialTransciever to allow the other protocols in the future? Just as a object I prefer to name it after what it is doing.